src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/CSharpSyntaxContext.cs

```diff
@@ -402,5 +402,43 @@ internal bool IsAwaitStatementContext(int position, CancellationToken cancellati
 
             return false;
         }
+
+        internal bool IsAwaitKeywordContext(int position)
+        {
+            if (IsGlobalStatementContext)
+            {
+                return true;
+            }
+
+            if (IsAnyExpressionContext || IsStatementContext)
+            {
+                foreach (var node in LeftToken.GetAncestors<SyntaxNode>())
+                {
+                    if (node.IsAnyLambdaOrAnonymousMethod())
+                    {
+                        return true;
+                    }
+
+                    if (node.IsKind(SyntaxKind.QueryExpression))
+                    {
+                        return false;
+                    }
+
+                    if (node.IsKind(SyntaxKind.LockStatement, out LockStatementSyntax? lockStatement))
+                    {
+                        if (lockStatement.Statement != null &&
+                            !lockStatement.Statement.IsMissing &&
```

Comment:
do you need the .IsMissing check?

Reply:
`lockStatement.Statement != null`.  This can't ever be true afaik.

Reply:
This is an exact copy from the old provider:

https://github.com/dotnet/roslyn/blob/1cca63b5d8ea170f8d8e88e1574aa3ebe354c23b/src/Features/CSharp/Portable/Completion/KeywordRecommenders/AwaitKeywordRecommender.cs#L43-L46

I'm not very confident, especially about the IsMissing check.

Reply:
ok.  leave in since this is a copy. :)